androidApp/src/main/java/com/example/tasky/android/agenda/viewmodel/AgendaDetailsViewModel.kt

```diff
@@ -53,11 +58,55 @@ class AgendaDetailsViewModel(
         )
     val screenStateFlow = _screenStateFlow.asStateFlow()
 
-    private val _isDeleteSuccess = MutableStateFlow(false)
-    val isDeleteSuccess = _isDeleteSuccess.asStateFlow()
+    private val _isDeleteSuccessFlow = MutableStateFlow(false)
+    val isDeleteSuccessFlow: StateFlow<UiEvent<Boolean>> =
+        _isDeleteSuccessFlow
+            .map {
+                object : UiEvent<Boolean> {
+                    override val data: Boolean
+                        get() = it
+                    override val onConsume: () -> Unit
+                        get() = {
+                            _isDeleteSuccessFlow.update { false }
+                        }
+                }
+            }.stateIn(
+                viewModelScope,
+                SharingStarted.Lazily,
+                object : UiEvent<Boolean> {
+                    override val data: Boolean
+                        get() = false
+                    override val onConsume: () -> Unit
+                        get() = {
+                            _isDeleteSuccessFlow.update { false }
+                        }
+                },
+            )
 
     private val _skippedImageCountFlow = MutableStateFlow(0)
-    val skippedImageCountFlow = _skippedImageCountFlow.asStateFlow()
+    val skippedImageCountFlow =
+        _skippedImageCountFlow
+            .map {
+                object : UiEvent<Int> {
+                    override val data: Int
+                        get() = it
```

Comment:
Same here, this logic can be encapsulated in another class in a sealed class (perhaps named `AgendaDetailsEvent`), which causes a snack bar to show on the screen when consumed. A much more fitting approach instead of modelling it as state, don’t you think?